llpc/patch/llpcPatchIntrinsicSimplify.h

```diff
@@ -0,0 +1,91 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2017-2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcPatchIntrinsicSimplify.h
+ * @brief LLPC header file: contains declaration of class Llpc::PatchIntrinsicSimplify.
+ ***********************************************************************************************************************
+ */
+#pragma once
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/IR/Intrinsics.h"
+
+#include "llpcPatch.h"
+#include "llpcPipelineState.h"
+
+namespace llvm
+{
+
+class AnalysisUsage;
+class Function;
+class FunctionPass;
+class Instruction;
+class IntrinsicInst;
+class ScalarEvolution;
+class Value;
+
+} // llvm
+
+namespace Llpc
+{
+
+class Context;
+
+// =====================================================================================================================
+// Represents the LLVM pass for intrinsic simplifications.
+class PatchIntrinsicSimplify final:
+    public llvm::FunctionPass
+{
+public:
+    explicit PatchIntrinsicSimplify();
+
+    void getAnalysisUsage(llvm::AnalysisUsage& analysisUsage) const override;
+    bool runOnFunction(llvm::Function& function) override;
```

Comment:
`getPassName` is not overridden. This is useful for debugging

Reply:
The pass name is also given in `INITIALIZE_PASS_BEGIN`, does `getPassName` have any additional effect?
Because if so, it should also be added to all other llpc passes.

Reply:
Ah, I didn't know it was given a default in pass initialization. The pass names show up were I need them in other LLPC passes, so all seems fine.